### PR TITLE
Drop legacy plexus API and use only JSR330 components

### PR DESCRIPTION
**Pull request by cstamas**
Drop:
- plexus container default
- legacy plexus APIs like Contextualizable
- drop Plexus Logger uses
- some minor cleanups for Java8 level
- converted Plexus XML to JSR330
- dropped plexus container (so no pcd or shim) as it is not needed anymore, plexus-io is not plx anymore as well

This is a breaking change (for example Plexus vs Slf4j logger use on APIs), so version bump is imminent.

### CODE REVIEW

**Comment by cstamas** [comment]
@sonatype-lift zip it

File: src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTBZ2FileResourceCollection.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import java.io.File;
+
+/**
+ * Alias for {@link PlexusIoTarBZip2FileResourceCollection}
+ */
+@Named( "tbz2" )
+public class PlexusIoTBZ2FileResourceCollection
+    extends PlexusIoTarBZip2FileResourceCollection
+{
+}
```
Comment (michael-o) [inline src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTBZ2FileResourceCollection.java:14]: `tgz` contains a new `newTarFile` method, this one does not. Why?

File: src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTarSnappyFileResourceCollection.java [hunk 2]
```diff
@@ -1,7 +1,11 @@
 package org.codehaus.plexus.archiver.tar;
 
+import javax.inject.Named;
+
 import java.io.File;
 
+// TODO: Name is okay? Seems not
+@Named( "snappy.gz" )
```
Comment (michael-o) [inline src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTarSnappyFileResourceCollection.java:8]: this look totally wrong. I'd expect `tar.snappy`

File: src/main/java/org/codehaus/plexus/archiver/tar/TarSnappyUnArchiver.java [hunk 3]
```diff
@@ -16,12 +16,15 @@
...
 
+import javax.inject.Named;
+
 import java.io.File;
 
 /**
  * Extract files in tar with snappy compression
  *
  */
+@Named( "tar.snappy" )
```
Comment (michael-o) [inline src/main/java/org/codehaus/plexus/archiver/tar/TarSnappyUnArchiver.java:27]: this one is correct

**Review by michael-o — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 4]
```diff
@@ -101,6 +108,36 @@
...
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest</artifactId>
+      <version>2.2</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <version>${slf4jVersion}</version>
```
[the remark this answers is not in the document]
Comment (cstamas) [reply pom.xml:127]: Fixed

File: src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTarSnappyFileResourceCollection.java (hunk 2 above)
Comment (cstamas) [reply src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTarSnappyFileResourceCollection.java:8]: All this comes from (under maintained) plexus XML file, there was even `par` vs `Par`
https://github.com/codehaus-plexus/plexus-archiver/blob/337214c2976564ff4255882bd76954c331a4e62e/src/main/resources/META-INF/plexus/components.xml#L183
https://github.com/codehaus-plexus/plexus-archiver/blob/337214c2976564ff4255882bd76954c331a4e62e/src/main/resources/META-INF/plexus/components.xml#L378

File: src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTBZ2FileResourceCollection.java (hunk 1 above)
Comment (cstamas) [reply src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTBZ2FileResourceCollection.java:14]: Found the leftover, removed

File: src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTarSnappyFileResourceCollection.java (hunk 2 above)
Comment (cstamas) [reply src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTarSnappyFileResourceCollection.java:8]: https://github.com/codehaus-plexus/plexus-archiver/blob/337214c2976564ff4255882bd76954c331a4e62e/src/main/resources/META-INF/plexus/components.xml#L405

Comment (michael-o) [reply src/main/java/org/codehaus/plexus/archiver/tar/PlexusIoTarSnappyFileResourceCollection.java]: :-D

File: src/main/java/org/codehaus/plexus/archiver/AbstractArchiver.java [hunk 5]
```diff
@@ -1314,14 +1281,7 @@ public void configureReproducibleBuild( FileTime lastModifiedTime )
...
         // 2. sort filenames in each directory when scanning filesystem
-        setFilenameComparator( new Comparator<String>()
-        {
-            @Override
-            public int compare( String s1, String s2 )
-            {
-                return s1.compareTo( s2 );
-            }
-        } );
+        setFilenameComparator( String::compareTo );
```
Comment (michael-o) [inline src/main/java/org/codehaus/plexus/archiver/AbstractArchiver.java:1284]: Nice!

**Review by michael-o — APPROVED** [review]

**Pull request merged**